File: src/js/_2591_Distribute_Money_to_Maximum_Children.js
/**
 * @param {number} money
 * @param {number} children
 * @return {number}
 */
var distMoney = function (money, children) {
    money -= children;

    if (money < 0)
        return -1;

    let childrenWith8 = Math.floor(money / 7);

    if (childrenWith8 == children && money % 7 == 0)
        return children;

    if (childrenWith8 == children - 1 && money % 7 == 3)
        return children - 2;

    if (childrenWith8 < children - 1)
        return childrenWith8;

    return children - 1;
};

File: src/python/_2591_Distribute_Money_to_Maximum_Children.py
class Solution:
    def distMoney(self, money: int, children: int) -> int:
        money -= children

        if money < 0:
            return -1

        if money // 7 == children and money % 7 == 0:
            return children

        if money // 7 == children - 1 and money % 7 == 3:
            return children - 2

        if money // 7 < children - 1:
            return money // 7

        return children - 1
